### ISSUE TITLE
import-keyword, module-keyword, and export-keyword are not preprocessing tokens

### ISSUE THREAD

**Issue by xmh0511** [cplusplus/draft#5301]
According to clause [lex.phases], phase 3 occurs previous to phase 4, In other words, the determination of preprocessing tokens occurs before the execution of preprocessing directives.

phase 3:
>The source file is decomposed into **preprocessing tokens** ([[lex.pptoken]](https://eel.is/c++draft/lex.pptoken)) and sequences of whitespace characters (including comments)[.](https://eel.is/c++draft/lex.phases#1.3.sentence-1)

phase 4:
> **Preprocessing directives are executed**, macro invocations are expanded, and _­Pragma unary operator expressions are executed.

That is, the replacement of preprocessing tokens *import* to *import-keyword*, *module* to *module-keyword*, and *export* to *export-keyword* all occur within the duration of execution of preprocessing directives. As specified in [cpp.module], and [cpp.import], they are:
> The module and export (if it exists) **preprocessing tokens** are replaced by the module-keyword and export-keyword preprocessing tokens respectively[.](https://eel.is/c++draft/cpp.module#3.sentence-1)
>
>In all three forms of [pp-import](https://eel.is/c++draft/cpp.import#nt:pp-import), the import and export (if it exists) **preprocessing tokens** are replaced by the import-keyword and export-keyword preprocessing tokens respectively[.](https://eel.is/c++draft/cpp.import#4.sentence-1)

The emphasized parts also confirm this point. We should admit that: *import*, *module*, and *export* are preprocessing tokens in phase 3. They also remain to be preprocessing tokens until the execution of the relevant preprocessing directives in phase 4. After these phases, these replaced *x-keywords* will be used as keywords in the subsequent phases.

**Comment by jensmaurer** [comment]
Or are you concerned that [lex.key] p1 only treats things in table 5 as keywords, but not the x-keyword things mentioned in the _keyword_ grammar?

**Comment by xmh0511** [comment]
You're correct on that the identifiers cover the original "import", "module", and "export" things. [lex.key] p1 should also cover the resulted  things from the former.

**Comment by AlisdairM** [comment]
Did Jens's response resolve this issue?
If not, what remains the concern?

**Comment by jensmaurer** [comment]
I think _import-keyword_ and friends are not necessarily identifiers, so are not in scope of [lex.key] p1 / table 5.
